### ISSUE TITLE
Midnight as closing hour

### ISSUE THREAD

**Issue by dlupu** [clemenst/time_of_day_attr#1]
Hello, 

I've been playing with this gem in the last minutes and, until now it perfectly fits my needs. Thank you for sharing it!

If I want a closing hour at midnight I would like that the value '24:00' be converted to 86400 and not to 0.

Also, for some reasons, '24:01', '24:02',  and so on until '24:59' are all converted to 0  or, in my opinion it should convert to nil since the format is not valid. 

Anyway, none of these issues are blocking for me , I just thought that you might find it useful to have some feedback.

Dorian

**Comment by dlupu** [comment]
fun fact : this is issue number #1 :+1:

**Comment by tlemens** [comment]
Hello Dorian,

thank you for opening the first issue and the feedback. I appreciate it.

I agree with you in all points. I will try to sort this out in 2015.

Best Regards

**Comment by tlemens** [comment]
'24:01' — '24:59' is currently not converted to nil, because Time.strptime ignores the minutes with format '%k'.

``` ruby
Time.strptime('24:01', '%k')
=> 2015-12-24 00:00:00 +0100

Time.strptime('24:01', '%k:%M')
ArgumentError: argument out of range
```

I would suggest to go with it and autoconvert '24:01' — '24:59' to '24'. What to you think, @dlupu ?

**Comment by dlupu** [comment]
Thanks adding support for the '24:00' format! 
For the rest, I tend to stick to my initial position : 24:01 should not be considered a valid value.
I would suggest to use a regexp to check that values are valid time.

I know this is pretty tricky because users can configure the time format. Thats why I would suggest to propose a default regexp that covers most cases and allow users to override the regexp if needed.

For instance `/\A((([ 01]?[0-9])|2[0-3])(\D{0,3})([0-5][0-9]))|(24(\D{0,3})00)\z/` covers the following cases:
- hours on left side of the separator,  minutes on the right side of the separators
- all values form 00:00 to 24:00 are supported
- hours from 0 to 9 with optional leading 0 or space
- optional separator of maximum 3 non-digit characters
  
  It's a bit complex but IMO pretty solid. WDYT?

**Comment by tlemens** [comment]
I like to keep it simple and only have one configuration (time formats).

But the setter now sets nil, when 'out of range' happened. So everything > '24:00' is no longer valid, as you suggested.

I just released version 1.0.0, which includes all fixed from this issue.

**Comment by dlupu** [comment]
That's nice! Thanks
